runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaCacheClientFetchFactory.java

```diff
@@ -1052,6 +1055,8 @@ private void doClientReplyBegin(
 
             this.initialGroupPartitionOffset = group.partitionOffset;
             this.initialGroupLatestOffset = group.latestOffset;
+            this.initialGroupStableOffset = group.stableOffset;
+            group.dataFrameReceived = false;
```

Comment:
We are attempting to detect whether or not a specific client has delivered a `DATA` frame, not the group.

If we track at the group level, then it will not behave differently for each client, which may have different filters, requiring different behavior.